File: bindings/python/examples/health_example.py
from orderflow import DataQualityFlags, Engine, EngineConfig, StreamKind, Symbol


def on_health(event: dict) -> None:
    print("health event:", event)


def main() -> None:
    cfg = EngineConfig(instance_id="py-health-example")
    with Engine(cfg) as engine:
        symbol = Symbol("CME", "ESM6", depth_levels=10)
        engine.subscribe(symbol, StreamKind.HEALTH, callback=on_health)
        engine.subscribe(symbol, StreamKind.ANALYTICS)

        engine.poll_once()
        engine.poll_once(DataQualityFlags.ADAPTER_DEGRADED)
        engine.poll_once()
        engine.unsubscribe(symbol)

        print("metrics:", engine.metrics())


if __name__ == "__main__":
    main()

File: bindings/java/src/main/java/com/orderflow/examples/HealthExample.java
package com.orderflow.examples;

import com.orderflow.bindings.DataQualityFlags;
import com.orderflow.bindings.EngineConfig;
import com.orderflow.bindings.OrderflowEngine;
import com.orderflow.bindings.OrderflowEvent;
import com.orderflow.bindings.StreamKind;
import com.orderflow.bindings.Symbol;

public final class HealthExample {
    private HealthExample() {}

    public static void main(String[] args) {
        EngineConfig defaults = EngineConfig.defaults();
        EngineConfig cfg = new EngineConfig(
                "java-health-example",
                defaults.configPath,
                defaults.logLevel,
                defaults.enablePersistence,
                defaults.auditMaxBytes,
                defaults.auditMaxFiles,
                defaults.auditRedactTokensCsv,
                defaults.dataRetentionMaxBytes,
                defaults.dataRetentionMaxAgeSecs);

        try (OrderflowEngine engine = new OrderflowEngine(null, cfg)) {
            engine.start();
            Symbol sym = new Symbol("CME", "ESM6", 10);
            engine.subscribe(sym, StreamKind.HEALTH, HealthExample::onHealthEvent);
            engine.subscribe(sym, StreamKind.ANALYTICS);

            engine.pollOnce(DataQualityFlags.NONE);
            engine.pollOnce(DataQualityFlags.ADAPTER_DEGRADED);
            engine.pollOnce(DataQualityFlags.NONE);
            engine.unsubscribe(sym);

            System.out.println("metrics=" + engine.metricsJson());
            engine.stop();
        }
    }

    private static void onHealthEvent(OrderflowEvent event) {
        System.out.println("health payload=" + event.payloadJson);
    }
}
